client/tx/tx.go

```diff
@@ -24,9 +25,85 @@ import (
 // or sign it and broadcast it returning an error upon failure.
 func GenerateOrBroadcastTxCLI(clientCtx client.Context, flagSet *pflag.FlagSet, msgs ...sdk.Msg) error {
 	txf := NewFactoryCLI(clientCtx, flagSet)
+
+	if flagAux, _ := flagSet.GetBool(flags.FlagAux); flagAux {
+		auxSignerData, err := makeAuxSignerData(clientCtx, txf, msgs...)
+		if err != nil {
+			return err
+		}
+		return clientCtx.PrintProto(&auxSignerData)
```

Comment:
I would move this whole block inside `GenerateOrBroadcastTxWithFactory`.

Also, I think instead of checking on the flag value directly, a better way would be to add a `clientCtx.IsAux`. Then inside `GenerateOrBroadcastTxWithFactory` do;

```go
if clientCtx.IsAux {
  makeAuxSignerData()
  printProto()
}
```